samples/bluetooth/unicast_audio_client/src/main.c

```diff
@@ -38,39 +38,154 @@ static K_SEM_DEFINE(sem_stream_qos, 0, 1);
 static K_SEM_DEFINE(sem_stream_enabled, 0, 1);
 static K_SEM_DEFINE(sem_stream_started, 0, 1);
 
-void print_hex(const uint8_t *ptr, size_t len)
+
+#if defined(CONFIG_LIBLC3CODEC)
+
+#include "lc3_config.h"
+#include "lc3_encoder.h"
+#include "math.h"
+
+#define AUDIO_SAMPLE_RATE_HZ    16000
+#define AUDIO_FREQUENCY_HZ      800
+#define AUDIO_LENGTH_US         10000
+#define NUM_SAMPLES             ((AUDIO_LENGTH_US * AUDIO_SAMPLE_RATE_HZ)/1000000)
+
+static int16_t audio_buf[NUM_SAMPLES];
+static Lc3Encoder_t lc3_encoder;
+
+/**
+ * Use the math lib to generate a sine-wave using 16 bit samples into a buffer.
+ *
+ * @param buf Destination buffer
+ * @param length_us Length of the buffer in microseconds
+ * @param frequency_hz frequency in Hz
+ * @param sample_rate_hz sample-rate in Hz.
+ */
+static void fill_audio_buf_sin(int16_t *buf, int length_us, int frequency_hz, int sample_rate_hz)
 {
-	while (len-- != 0) {
-		printk("%02x", *ptr++);
+	int sine_period = sample_rate_hz/frequency_hz;
+	int num_samples = (length_us * sample_rate_hz) / 1000000;
+	float step = 3.1415 / sine_period;
+	float sample;
+
+	for (int i = 0; i < num_samples; i++) {
+		sample = sin(i * step);
+		buf[i] = (int16_t)(INT16_MAX * sample);
+		/* printk("%3i: rad: %i sample: %i %6i\n", i, (int)(rad*1000), (int)(sample*1000),
+		 * (int16_t)(sample * INT16_MAX));
+		 */
 	}
 }
 
-static void print_codec(const struct bt_codec *codec)
+static const char *getErrString(uint16_t err)
 {
-	printk("codec 0x%02x cid 0x%04x vid 0x%04x count %u\n",
-	       codec->id, codec->cid, codec->vid, codec->data_count);
+	switch (err) {
+	case LC3_ENCODE_ERR_FREE:
+		return "LC3_ENCODE_ERR_FREE";
+	case LC3_ENCODE_ERR_INVALID_CONFIGURATION:
+		return "LC3_ENCODE_ERR_INVALID_CONFIGURATION";
+	case LC3_ENCODE_ERR_INVALID_BYTE_COUNT:
+		return "LC3_ENCODE_ERR_INVALID_BYTE_COUNT";
+	case LC3_ENCODE_ERR_INVALID_BITS_PER_AUDIO_SAMPLE:
+		return "LC3_ENCODE_ERR_INVALID_BITS_PER_AUDIO_SAMPLE";
+	case LC3_ENCODE_ERR_ENCODER_ALLOCATION:
+		return "LC3_ENCODE_ERR_ENCODER_ALLOCATION";
+	}
+	return "UNKNOWN LC3 ENCODE ERROR";
+}
 
-	for (size_t i = 0; i < codec->data_count; i++) {
-		printk("data #%zu: type 0x%02x len %u\n",
-		       i, codec->data[i].data.type,
-		       codec->data[i].data.data_len);
-		print_hex(codec->data[i].data.data,
-			  codec->data[i].data.data_len -
-				sizeof(codec->data[i].data.type));
-		printk("\n");
+static void lc3_audio_timer_timeout(struct k_work *work)
+{
+	/* For the first call-back we push multiple audio frames to the buffer to use the
+	 * controller ISO buffer to handle jitter.
+	 */
+	static int prime_count = 2;
```

Comment:
Looks like this doesn't need to be static, but const (and unsigned). 

```suggestion
	const uint8_t prime_count = 2;
```

Reply:
Correct, thanks